smart_lists/mixins.py

```diff
@@ -24,8 +27,40 @@ def get_queryset(self):
         if filters:
             for fltr in filters:
                 qs = qs.filter(**fltr)
+        qs = self.get_search_results(qs)
         return qs
 
+    def get_search_results(self, queryset):
+        """
+        borrowed from django-admin
+        @param queryset:
+        @return: filtered queryset
+        """
+        search_term = self.request.GET.get(self.search_query_parameter_name, None)
+
+        if search_term is None:
+            return queryset
+        # Apply keyword searches.
+        def construct_search(field_name):
+            if field_name.startswith('^'):
```

Comment:
Re-implement search lookup names similar to django admin, not use strange looking characters that might have weird URL encodings.

Query should look like this:
```?q=asd```

Variables should look like this:
```search_fields = ('company_name', 'code')```